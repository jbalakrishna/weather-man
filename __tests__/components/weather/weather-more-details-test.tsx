import WeatherMoreDetails from "@/components/weather/weather-more-details";
import { render, waitFor } from "@testing-library/react-native";

describe("weather details ", () => {
  test("weather details with static details renders correctly", async () => {
    const tree = render(
      <WeatherMoreDetails
        humidity={22}
        dewpoint={"6"}
        wind={"20"}
        windchill={"10"}
        cloud={5}
      />
    ).toJSON();

    await waitFor(() => {
      expect(tree).toMatchSnapshot();
    });
  });
});
