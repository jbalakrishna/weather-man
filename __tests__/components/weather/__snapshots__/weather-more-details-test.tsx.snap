// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weather details  weather details with static details renders correctly 1`] = `
[
  <Text
    className="font-bold text-2xl text-slate-700"
  >
    Details
  </Text>,
  <View
    className="flex-1 gap-8 px-5 py-8 bg-slate-100 rounded-xl"
  >
    <View
      className="flex-1 flex-row gap-4 justify-between"
    >
      <Text
        className="font-semibold text-slate-400 text-xl"
      >
        Humidity  
        <Text
          className="text-2xl font-bold text-slate-600"
        >
          22%
        </Text>
      </Text>
      <Text
        className="font-semibold text-slate-400 text-xl"
      >
        Dew  
        <Text
          className="text-2xl font-bold text-slate-600"
        >
          6
        </Text>
      </Text>
    </View>
    <View
      className="flex-1 flex-row justify-between"
    >
      <Text
        className="font-semibold text-slate-400 text-xl"
      >
        Wind  
        <Text
          className="text-2xl font-bold text-slate-600"
        >
          20
        </Text>
      </Text>
      <Text
        className="font-semibold text-slate-400 text-xl"
      >
        Wind chill  
        <Text
          className="text-2xl font-bold text-slate-600"
        >
          10
        </Text>
      </Text>
    </View>
    <View
      className="flex-1 flex-row justify-end"
    >
      <Text
        className="font-semibold text-slate-400 text-xl"
      >
        Clouds  
        <Text
          className="text-2xl font-bold text-slate-600"
        >
          5%
        </Text>
      </Text>
    </View>
  </View>,
]
`;
